Lesson3/ViewController.swift

```diff
@@ -27,69 +32,53 @@ class ViewController: UIViewController {
-    func onDataChange(student: Student) {
-        students.append(student)
-        tableView.reloadData()
-    }
-}
-
-extension ViewController: TableViewCellDelegate {
-    func didTapButton(student: Student, cell: UITableViewCell) {
-        guard let cellIndexPath = tableView.indexPath(for: cell) else { return }
-        students.remove(at: cellIndexPath.row)
-        students.insert(student, at: cellIndexPath.row)
-        tableView.reloadRows(at: [cellIndexPath], with: .bottom)
-    }
-}
-  
 // MARK: - Table view data source & delegate
     
 extension ViewController: UITableViewDataSource, UITableViewDelegate {
 
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return students.count
+        return posts.count
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "cellIdentifier", for: indexPath) as! TableViewCell
+        let post = posts[indexPath.row]
+        if (post.content != nil) && (post.contentPicture != nil){
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: "TextAndImageTableViewCell", for: indexPath) as? TextAndImageTableViewCell  else { return UITableViewCell() }
+            cell.configure(post: post)
+            return cell
+        } else if post.content != nil {
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: "TextTableViewCell", for: indexPath) as? TextTableViewCell else { return UITableViewCell() }
+            cell.configure(post: post)
+            return cell
+        } else {
+            guard let cell = tableView.dequeueReusableCell(withIdentifier: "ImageTableViewCell", for: indexPath) as? ImageTableViewCell else { return UITableViewCell() }
+            cell.configure(post: post)
+            return cell
+        }
         
-        cell.configure(student: students[indexPath.row], delegate: self)
-
-        return cell
     }
     
     func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-        return 100
+        let post = posts[indexPath.row]
+        
+        if (post.content != nil) && (post.contentPicture != nil){
+            return 400
+        }
+        if post.content != nil{
+            return 100
+        }
+        return 300
     }
     
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        let student = students[indexPath.row]
+        tableView.deselectRow(at: indexPath, animated: true)
+        let post = posts[indexPath.row]
         let sb = UIStoryboard(name: "Main", bundle: nil)
-        let vc = sb.instantiateViewController(identifier: "StudentDetailViewController") as! StudentDetailViewController
-        vc.student = student
+        let vc = sb.instantiateViewController(identifier: "PostDetailViewController") as! PostDetailViewController
```

Comment:
```suggestion
        let viewController = sb.instantiateViewController(identifier: "PostDetailViewController") as! PostDetailViewController
```
лучше не оставлять как было